WEBSOCKETS.md

```diff
@@ -115,6 +119,22 @@ stateDiagram-v2
     L --> [*]
 ```
 
+If the server sends a load dataset command, because of network latency, it is possible for the clients to send other commands of their own while not having received the load command.  
+In this case, the server will buffer these commands and replay them after the loading is done.
```

Comment:
Considering this for implementation, I came to the conclusion that such handling would introduce additional complexity on server side, though with no compelling argument why it would be really necessary:

Most commands actually probably don't make sense to be replayed after a new dataset has been loaded - any snapshot or object message should probably be ignored (default object parameters will be applied after dataset loading; snapshots of the "old" object also probably should be discarded after loading a new dataset). a different load dataset message received at the same time also should probably better be discarded (otherwise, the current dataset loading procedure should actually better be cancelled instead).

Reply:
TLDR: I would stick with the current implementation on the server: discarding messages coming in during that time. Maybe some feedback to the client could help in discovering potential resulting misunderstandings.

Reply:
I think you're right, we should keep the current implementation.
If we change datasets successfully, the state of objects is already reset, so they are always in sync.

However, one case in which it would make sense to NAK the clients would be the following:
The server wants to change to a dataset, that a client doesn't have.
The client simultaneously sends a object manipulation command (which reaches the server after dataset load was sent --> message is discarded).
The client answers with NAK (it doesn't have the dataset).

Now we have two states:
The server in the state before the object manipulation command (because discarded).
The client after manipulating the object (the command doesn't require acknowledgement).

One way to resolve this, could be to "mark" clients which have sent commands while in the dataset loading process, and in case they NAK, send the state of the server to these clients (objects positions, snapshots, etc.).
What do you think about it?